packages/vue-pdf-viewer/src/components/PDFViewer.vue

```diff
@@ -153,13 +155,32 @@ export default {
             this.pdfViewer.currentScaleValue = this.currentScale
         })
 
-        const documentLoadingTask = pdfjsLib.getDocument({
-            url: this.url,
+        const docOptions = {
             cMapUrl: this.pdfjsCMapUrl,
             cMapPacked: true
-        })
+        }
+
+        if (this.source.constructor === Uint8Array) {
+            docOptions.data = this.source
+        } else {
+            docOptions.url = this.source
+        }
+
+        const documentLoadingTask = pdfjsLib.getDocument(docOptions)
         this.pdfDocument = await documentLoadingTask.promise
 
+        eventBus.on('documentloaded', function() {
+            const event = document.createEvent('Event')
+            event.initEvent('PDFViewer:documentLoaded', true, true)
+            window.dispatchEvent(event)
+        })
+
+        eventBus.on('pagesloaded', function() {
+            const event = document.createEvent('Event')
+            event.initEvent('PDFViewer:pagesLoaded', true, true)
+            window.dispatchEvent(event)
+        })
```

Comment:
According to the mdn web docs this is the "old fashioned way" of creating custom events:
https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events#the_old-fashioned_way

Reply:
Oh yes, adjusted it